source/core/parameters.cc

```diff
@@ -2200,8 +2199,65 @@ namespace Parameters
           std::string shape_arguments_str = prm.get("shape arguments");
           std::vector<std::string> shape_arguments_str_list(
             Utilities::split_string_list(shape_arguments_str, ";"));
-          std::vector<double> shape_arguments =
-            Utilities::string_to_double(shape_arguments_str_list);
+          std::vector<double> shape_arguments;
+          if (shape_type == "rbf")
+            {
+              if constexpr (dim == 3)
+                {
+                  std::string shape_name = shape_arguments_str_list[0];
+                  std::map<std::string, std::vector<double>> rbf_data;
```

Comment:
I don't understand. When you declare a default entry, why are you filling from a file like this? It does not make sense no?
I feel like this bunch of code is repeated lower when you actually do the parsing?

Reply:
I removed the code duplication and added the default case